albert-masats/precurso/tema3/bingo.js

```diff
@@ -0,0 +1,210 @@
+    {number: pick15FromRanArray(), matched: false },
+    {number: pick15FromRanArray(), matched: false },
+    {number: pick15FromRanArray(), matched: false },
+    {number: pick15FromRanArray(), matched: false },
+    ];
+    return bingoCard;
+}   
+
+function bingoCardView() { //Vista para prompt
+    
+    alert(`                                       CARTÓN:
+                    ${bingoCard[0].number}     ${bingoCard[1].number}     ${bingoCard[2].number}     ${bingoCard[3].number}     ${bingoCard[4].number}
+                    ${bingoCard[5].number}     ${bingoCard[6].number}     ${bingoCard[7].number}     ${bingoCard[8].number}     ${bingoCard[9].number}
+                    ${bingoCard[10].number}     ${bingoCard[11].number}     ${bingoCard[12].number}     ${bingoCard[13].number}     ${bingoCard[14].number}`);
+}
+
+function bingoCardConsole() {       //Vista consola
+    console.log('Primera Linea');console.log(bingoCard[0].number+'   '+bingoCard[1].number+'   '+bingoCard[2].number+'   '+bingoCard[3].number+'   '+bingoCard[4].number);
+    console.log('Segunda Linea');
+    console.log(bingoCard[5].number+'   '+bingoCard[6].number+'   '+bingoCard[7].number+'   '+bingoCard[8].number+'   '+bingoCard[9].number);
+    console.log('Tercera Linea');
+    console.log(bingoCard[10].number+'   '+bingoCard[11].number+'   '+bingoCard[12].number+'   '+bingoCard[13].number+'   '+bingoCard[14].number);
+}
+
+function questionCard() {
+    yourCard = prompt('¿Quieres este carton? Responde:yes (para aceptarlo), Responde: no (para rechazarlo)');
+    while (yourCard !== 'yes' && yourCard !== 'no' && yourCard !== null) {
+        yourCard = prompt('¿Quieres este carton? Responde:yes (para aceptarlo), Responde: no (para rechazarlo)');
+    }
+    if (yourCard === null) {
+        return('bye');
+    }else if(yourCard === 'no') {
+        newCard();
+    }else{
+        alert('¡Perfecto, empezemos!');
+    }
+}
+
+function turns() {
+    bingoCounter = 0;
+    totalPoints = 900;
+
+    for (let i=0; i<numbersHat.length; i++) { 
+        for (let j=0; j<bingoCard.length; j++) {
+            if (numbersHat[i] === bingoCard[j].number) {
+                bingoCard[j].matched = true;
+                bingoCard[j].number = 'X';
+            }
+        }
+        bingoCounter += 1;
+        console.log(numbersHat[i]);
+        alert(numbersHat[i]);
+        totalPoints -= 10;
+        const ALL_MATCHED = bingoCard.every(cardNumber => cardNumber.matched === true);
+        if(ALL_MATCHED === true) {
+            return alert('¡¡¡BINGO!!!');
+        }
+        if (lineConfirmation === false) {
+            let firstLineCount = 0;
+            let secondLineCount = 0;
+            let thirdLineCount =0;
+            for(k=0; k<bingoCard.length -10; k++) {
+                if(bingoCard[k].matched === true) {
+                    firstLineCount++;
+                }
+            }
+            for(l=5; l<bingoCard.length -5; l++) {
```

Comment:
Si has pasado el sonarqube ya lo habrás visto. Tenemos que iniciar la varibale del for --> for(**let** l=5; ...)